docs/reference/rest-api/rest-api-compatibility.asciidoc

```diff
@@ -0,0 +1,72 @@
+[[rest-api-compatibility]]
```

Comment:
I could use some help figuring out how to link up a couple TODO's below. Also, I left the line lengths long here in anticipation of many edits. I will reduce the length prior to merge.